system/forcefield.py

```diff
@@ -18,16 +18,16 @@ def merge_potentials(nrgs):
     for a in nrgs:
         a_name = a[0]
         a_args = a[1]
-        if a_name == custom_ops.HarmonicBond_f32:
-            c_nrgs.append(custom_ops.HarmonicBond_f32(a_args[0], a_args[1]))
-        elif a_name == custom_ops.HarmonicAngle_f32:
-            c_nrgs.append(custom_ops.HarmonicAngle_f32(a_args[0], a_args[1]))
-        elif a_name == custom_ops.PeriodicTorsion_f32:
-            c_nrgs.append(custom_ops.PeriodicTorsion_f32(a_args[0], a_args[1]))
-        elif a_name == custom_ops.LennardJones_f32:
-            c_nrgs.append(custom_ops.LennardJones_f32(a_args[0].astype(np.float32), a_args[1]))
-        elif a_name == custom_ops.Electrostatics_f32:
-            c_nrgs.append(custom_ops.Electrostatics_f32(a_args[0].astype(np.float32), a_args[1]))
+        if a_name == custom_ops.HarmonicBond_f64:
```

Comment:
I take it the training was only stable under 64bit precision? Did we ever figure out why it didn't work in 32 bit?

Reply:
I recall the gradients blowing up in 32-bit precision, but I didn't look too deeply into the specific numerical error